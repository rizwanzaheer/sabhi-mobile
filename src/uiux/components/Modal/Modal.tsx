import * as React from 'react'
import Container from '../Container/Container'
import { ScrollView } from 'react-native'
import { withTheme } from '../../theming'
import Button from '../Button/Button'
import Icon from '../Icon/Icon'
// import Device from '../../services/device'
import { View, LayoutChangeEvent } from 'react-native'
import useLayout from '../../hooks/useLayout'
import { Device } from '@uiux'

interface ModalProps {
  /**
   * Enable scrollView
   */
  scrollEnabled?: boolean

  /**
   * Dismiss the modal view
   */
  dismiss?: () => void

  /**
   * Theme HOC
   */
  theme: any
}

const Modal: React.FC<ModalProps> = props => {
  const [layout, updateLayout] = useLayout()
  const onLayout = (event: LayoutChangeEvent) => {
    updateLayout(event.nativeEvent.layout)
  }
  const scrollViewContent = <ScrollView>{props.children}</ScrollView>
  const tabletSplitView = layout && layout.width > 700
  const tabletFullScreenStyle = {
    width: 600,
    height: layout && layout.height / 1.5,
    marginTop: 100,
    paddingBottom: 10,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
  }

  return (
    <View style={{ flex: 1 }} onLayout={onLayout}>
      <Container
        marginTop={50}
        marginLeft={10}
        marginRight={10}
        background={'secondary'}
        viewStyle={{
          alignSelf: tabletSplitView ? 'center' : 'stretch',
          elevation: 2,
          shadowColor: '#000000',
          shadowRadius: 20,
          shadowOpacity: 0.2,
          borderTopLeftRadius: 10,
          borderTopRightRadius: 10,
          ...(tabletSplitView ? tabletFullScreenStyle : { flex: 1 }),
        }}
      >
        <Container alignItems={'center'} justifyContent={'center'} padding={8}>
          {Device.isIOS && (
            <Container br={5} backgroundColor={props.theme.colors.primary.accessories} w={50} h={5} />
          )}
        </Container>
        {props.dismiss && (
          <Container viewStyle={{ position: 'absolute', right: 10, top: 10, zIndex: 10 }}>
            <Button
              iconButton={true}
              icon={
                <Icon icon={props.theme.icons.CLOSE} color={props.theme.colors.primary.accessory} size={40} />
              }
              onPress={props.dismiss}
            />
          </Container>
        )}
        {props.scrollEnabled ? scrollViewContent : props.children}
      </Container>
    </View>
  )
}

export default withTheme(Modal)
