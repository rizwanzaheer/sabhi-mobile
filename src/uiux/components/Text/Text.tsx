import * as React from 'react'
import { Text, TextStyle } from 'react-native'
import { withTheme } from '../../theming'
import * as Kancha from '../../types'

export const TextTypes: Kancha.TextTypesStatic = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
  H5: 'h5',
  ListItem: 'listItem',
  ListItemRight: 'listItemRight',
  ListItemNote: 'listItemNote',
  SubTitle: 'subTitle',
  ActivityTitle: 'activityTitle',
  ActivitySubTitle: 'activitySubTitle',
  Body: 'body',
  Button: 'button',
  ButtonSmall: 'buttonSmall',
  NavButton: 'navButton',
  Summary: 'summary',
  SectionHeader: 'sectionHeader',
}
/**
 * Kancha Text Props
 */

export interface TextProps {
  /**
   * A testID for the text element
   */
  testID?: string
  /**
   * The type of text to display. This will be styled accordinly to the theme
   */
  type?: string

  /**
   * The type of text to display. This will be styled accordinly to the theme
   */
  onPress?: () => void

  /**
   * Overide the color with a warning color
   */
  warn?: boolean

  /**
   * Color prop is used to configure button text colors
   */
  buttonTextColor?: Kancha.BrandPropOptions

  /**
   * Overide the brand color
   */
  textColor?: string

  /**
   * Overide the color with a warning color
   */
  block?: Kancha.BlockPropsOptions

  /**
   * Make the text bold
   */
  bold?: boolean

  /**
   * The padding around the text
   */
  padding?: number

  /**
   * A bottom padding for the text. Useful for headings
   */
  paddingBottom?: number | boolean | undefined

  /**
   * The margin around the text
   */
  margin?: number

  /**
   * The margin around the text
   */
  textAlign?: string

  /**
   * Decoration for button text
   */
  textDecorationLine?: 'none' | 'underline' | 'line-through' | 'underline line-through' | undefined

  /**
   * Transform the text
   */
  transform?: 'uppercase' | 'lowercase' | undefined

  /**
   * Transform the text
   */
  textStyle?: TextStyle

  /**
   * All the text to be selectable
   */
  selectable?: boolean

  /**
   * Transform the text
   */
  theme: any
}

const KanchaText: React.FC<TextProps> & { Types: Kancha.TextTypesStatic } = props => {
  const TextThemeMap: Kancha.TextDefaultThemeMapTypes = {
    h1: {
      fontSize: props.theme.text.sizes.h1,
      color: props.theme.colors.primary.text,
    },
    h2: {
      fontSize: props.theme.text.sizes.h2,
      color: props.theme.colors.primary.text,
    },
    h3: {
      fontSize: props.theme.text.sizes.h3,
      color: props.theme.colors.primary.text,
    },
    h4: {
      fontSize: props.theme.text.sizes.h4,
      color: props.theme.colors.primary.text,
    },
    h5: {
      fontSize: props.theme.text.sizes.h5,
      color: props.theme.colors.primary.text,
    },
    activityTitle: {
      fontSize: props.theme.text.sizes.activityTitle,
      color: props.theme.colors.primary.text,
    },
    activitySubTitle: {
      fontSize: props.theme.text.sizes.activitySubTitle,
      color: props.theme.colors.secondary.text,
    },
    subTitle: {
      fontSize: props.theme.text.sizes.subTitle,
      color: props.theme.colors.secondary.text,
    },
    listItem: {
      fontSize: props.theme.text.sizes.listItem,
      color: props.theme.colors.primary.text,
    },
    listItemNote: {
      fontSize: props.theme.text.sizes.listItemNote,
      color: props.theme.colors.secondary.text,
    },
    listItemRight: {
      fontSize: props.theme.text.sizes.listItemRight,
      color: props.theme.colors.secondary.text,
    },
    summary: {
      fontSize: props.theme.text.sizes.summary,
      color: props.theme.colors.secondary.text,
    },
    body: {
      fontSize: props.theme.text.sizes.body,
      color: props.theme.colors.primary.text,
      lineHeight: props.theme.text.lineHeights.body,
    },
    button: {
      fontSize: props.theme.text.sizes.button,
    },
    buttonSmall: {
      fontSize: props.theme.text.sizes.buttonSmall,
    },
    navButton: {
      fontSize: props.theme.text.sizes.navButton,
    },
    sectionHeader: {
      fontSize: props.theme.text.sizes.sectionHeader,
      color: props.theme.colors.secondary.text,
    },
  }
  const styles: TextStyle = {
    ...(props.type ? { ...TextThemeMap[props.type] } : { ...TextThemeMap[TextThemeMap.body] }),
    ...(props.textColor ? { color: props.textColor } : {}),
    ...(props.bold ? { fontWeight: 'bold' } : {}),
    ...(props.warn ? { color: props.theme.colors.theme.WARN } : {}),
    ...(props.textAlign ? { textAlign: props.textAlign } : {}),
    ...(props.buttonTextColor
      ? {
          color: props.block
            ? props.theme.colors[props.buttonTextColor].buttonText[props.block]
            : props.theme.colors[props.buttonTextColor].buttonText.filled,
        }
      : {}),
    ...(props.paddingBottom ? { paddingBottom: props.paddingBottom } : {}),
    ...(props.paddingBottom && typeof props.paddingBottom === 'boolean'
      ? { paddingBottom: props.theme.spacing.default }
      : {}),
    ...(props.transform ? { textTransform: props.transform } : {}),
    ...(props.textStyle ? { ...props.textStyle } : {}),
    flexWrap: 'wrap',
  }

  return (
    <Text
      selectable={props.selectable}
      style={styles}
      testID={props.testID}
      accessibilityLabel={props.testID}
      onPress={props.onPress && props.onPress}
    >
      {props.children}
    </Text>
  )
}

KanchaText.Types = TextTypes

export default withTheme(KanchaText)
