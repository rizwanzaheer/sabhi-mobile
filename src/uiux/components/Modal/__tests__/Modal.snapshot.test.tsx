import * as React from 'react'
import { render, act } from 'react-native-testing-library'
import Modal from '../Modal'
import Text from '../../Text/Text'

describe('Component(snapshots): Modal', () => {
  it('should render with no props', () => {
    const tree = render(<Modal />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with props', () => {
    const dismiss = jest.fn()

    const tree = render(<Modal scrollEnabled={true} dismiss={dismiss} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with props and children', () => {
    const dismiss = jest.fn()

    const tree = render(
      <Modal scrollEnabled={true} dismiss={dismiss}>
        <Text>Test Modal</Text>
      </Modal>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render layout ', () => {
    // mock nativeEvent with desired dimensions
    const mockNativeEvent = {
      nativeEvent: {
        layout: {
          width: 820,
        },
      },
    }
    const component = render(<Modal scrollEnabled={true} />)
    let tree = component.toJSON()

    act(() => {
      if (tree) {
        tree.props.onLayout(mockNativeEvent)
      }
    })
    tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
