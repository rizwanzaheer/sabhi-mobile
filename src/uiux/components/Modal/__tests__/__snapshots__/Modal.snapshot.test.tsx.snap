// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): Modal should render layout  1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "alignSelf": "center",
        "backgroundColor": "#EFEEF3",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderRadius": undefined,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": undefined,
        "flexDirection": undefined,
        "height": NaN,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 100,
        "padding": undefined,
        "paddingBottom": 10,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000000",
        "shadowOpacity": 0.2,
        "shadowRadius": 20,
        "width": 600,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": 8,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#BBBBBB",
            "borderRadius": 5,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 5,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": 50,
          }
        }
      />
    </View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`Component(snapshots): Modal should render with no props 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "alignSelf": "stretch",
        "backgroundColor": "#EFEEF3",
        "borderRadius": undefined,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 50,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000000",
        "shadowOpacity": 0.2,
        "shadowRadius": 20,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": 8,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#BBBBBB",
            "borderRadius": 5,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 5,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": 50,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Component(snapshots): Modal should render with props 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "alignSelf": "stretch",
        "backgroundColor": "#EFEEF3",
        "borderRadius": undefined,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 50,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000000",
        "shadowOpacity": 0.2,
        "shadowRadius": 20,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": 8,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#BBBBBB",
            "borderRadius": 5,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 5,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": 50,
          }
        }
      />
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": undefined,
          "zIndex": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 8,
            "maxWidth": 300,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": "center",
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": 10,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Icon
              color="#BBBBBB"
              name="ios-close"
              size={40}
            />
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": "center",
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`Component(snapshots): Modal should render with props and children 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "alignSelf": "stretch",
        "backgroundColor": "#EFEEF3",
        "borderRadius": undefined,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 2,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 50,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000000",
        "shadowOpacity": 0.2,
        "shadowRadius": 20,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": 8,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#BBBBBB",
            "borderRadius": 5,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 5,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": 50,
          }
        }
      />
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": undefined,
          "zIndex": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 8,
            "maxWidth": 300,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": "center",
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": 10,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Icon
              color="#BBBBBB"
              name="ios-close"
              size={40}
            />
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": "center",
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView>
      <View>
        <Text
          style={
            Object {
              "flexWrap": "wrap",
            }
          }
        >
          Test Modal
        </Text>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
